doc/design/ops/rnn.md

```diff
@@ -0,0 +1,126 @@
+# RNN design
+This is the design doc of the recurrent neural network operator, 
+and this operator requires that input instances in each mini-batch must have the same length. 
+
+## RNN Algorithm Implementation
+
+<p aligh="center">
+<img src="./images/rnn.jpg"/>
+</p>
+
+The above diagram shows an RNN being unrolled into a full network.
+
+There are several important concepts:
+
+- stepnet, the network execute in every time step 
+- memory, a variable storing state in the current step, which is denoted as $h_t$.
+- pre-memory, the value of state in the previous time step, it can be denoted as $h_{t-1}$.
+- init-memory, the variable to help initialize state in the first time step.
+
+### step scopes
+
```

Comment:
The step-net could have local variables defined. In each step of RNN execution, a scope is created to hold corresponding variables.  Such a scope is known as a *step scope*.